code/datums/gamemodes/mixed.dm

```diff
@@ -135,6 +142,10 @@
 				traitors += tplayer
 				token_players.Remove(tplayer)
 				tplayer.special_role = ROLE_ARCFIEND
+			if(ROLE_SALVAGER)
```

Comment:
This won't work unless you add `ROLE_SALVAGER` to the `traitor_types` list above.